servant/src/Servant/API/Times.hs

```diff
@@ -0,0 +1,105 @@
+{-# LANGUAGE CPP #-}
+{-# LANGUAGE DataKinds          #-}
+{-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE PolyKinds          #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Servant.API.Times
+    ( FTime(..)
+    , toProxy
+    , getFormat
+    , renderTime
+    , parseTime
+    ) where
+
+import           Data.Typeable (Typeable)
+import           GHC.TypeLits  -- (Symbol)
+import           Web.HttpApiData
+import qualified Data.Time.Format as T
+#if !MIN_VERSION_time(1,5,0)
+import           System.Locale as T
+#endif
+import           Data.Text (pack, Text)
+import           Data.Proxy
+import           Control.Monad ((>=>))
+import           Control.Arrow (first)
+
+-- | A wrapper around a time type which can be parsed/rendered to with `format',
+-- as specified in 'Data.Time.Format'.
+--
+-- Example:
+-- >>>            -- GET /events/:date
+-- >>> type MyApi = "events" :> Capture "date" (FTime "%Y-%m-%d" Day) :> Get '[JSON] [Event]
+newtype FTime (format :: Symbol) t = FTime {getFTime :: t}
+    deriving (Typeable, Eq, Ord)
+
+instance (KnownSymbol format, T.FormatTime t) => Show (FTime format t) where
+    showsPrec i t = showParen (i > 1) (\str -> renderTime t ++ str)
```

Comment:
I wanted the Read and Show instances to use the format that their type specified, also happens to make playing with it in say GHCi a bit easier (and should keep the `show . read` identity)